packages/zoltan2/test/color/coloring1.cpp

```diff
@@ -197,31 +255,97 @@ int main(int narg, char** arg)
   //int balanceColors = 0;
   cmdp.setOption("color_choice", &colorMethod,
        "Color choice method: FirstFit, LeastUsed, Random, RandomFast");
+  int batchSize = -1;
+  cmdp.setOption("batchSize", &batchSize,
+               "Batch Size for distributed coloring. Negative signifies all.");
+  std::string kokkosOnlyInterior("false");
+  cmdp.setOption("kokkosOnlyInterior", &kokkosOnlyInterior,
+               "Sets whether Kokkos colors only interior vertices (true, false)");
   // cmdp.setOption("balance_colors", &balanceColors,
   //                "Balance the size of color classes: 0/1 for false/true");
 
   //////////////////////////////////
   cmdp.parse(narg, arg);
+  if (prepartition != "") {
+    if (prepartition == "rows") prepartition_rows = true;
+    else if (prepartition == "nonzeros") prepartition_nonzeros = true;
+    else {
+      std::cout << "Invalid value of prepartition option " << prepartition
+                << std::endl;
+      std::cout << "No prepartitioning will be done" << std::endl;
+      std::cout << "FAIL" << std::endl;
+      return -1;
+    }
+  }
 
   RCP<UserInputForTests> uinput;
-
-  if (inputFile != ""){ // Input file specified; read a matrix
-    uinput = rcp(new UserInputForTests(testDataFilePath, inputFile, comm, true));
+  RCP<SparseMatrix> Matrix;
+  if(inputFile.find("bin") != string::npos){//we're dealing with a binary file
+    Matrix = readBinaryFile<SparseMatrix>(inputFile,comm,true,false);
+  } else { 
+    if (inputFile != ""){ // Input file specified; read a matrix
+      uinput = rcp(new UserInputForTests(testDataFilePath, inputFile,
+                                         comm, true));
+      printf("--Rank %d: Successfully read the file\n",comm->getRank());
+    }
+    else {                 // Let Galeri generate a matrix
+      uinput = rcp(new UserInputForTests(xdim, ydim, zdim, matrixType,
+                                         comm, true, true));
+    }
+    Matrix = uinput->getUITpetraCrsMatrix();
   }
-  else                  // Let Galeri generate a matrix
-
-    uinput = rcp(new UserInputForTests(xdim, ydim, zdim, matrixType, comm, true, true));
-
-  RCP<SparseMatrix> Matrix = uinput->getUITpetraCrsMatrix();
-
   if (me == 0)
     std::cout << "NumRows     = " << Matrix->getGlobalNumRows() << std::endl
          << "NumNonzeros = " << Matrix->getGlobalNumEntries() << std::endl
          << "NumProcs = " << comm->getSize() << std::endl;
 
+  if (prepartition != "") {
+
+    // Compute new partition of matrix
+    SparseMatrixAdapter *zadapter;
```

Comment:
We have `std::unique_ptr`.  Don't use raw pointers; use `unique_ptr`.

Reply:
Done